Sally7/S7ConnectionHelpers.cs

```diff
@@ -59,8 +59,8 @@ public static int BuildReadRequest(in Span<byte> buffer, in IReadOnlyList<IDataI
         public static int BuildWriteRequest(in Span<byte> buffer, in IReadOnlyList<IDataItem> dataItems)
         {
             var span = buffer.Slice(17); // Skip header
-            span[0] = (byte) FunctionCode.Write;
             span[1] = (byte) dataItems.Count;
+            span[0] = (byte) FunctionCode.Write;
```

Comment:
This is just a mini-optimization. By flipping the order (the largest index first), the JIT only needs to emit one bound check instead of two.